### ISSUE TITLE
lxd snap 4 LTS daemon fails to start if network is slow to start

### ISSUE THREAD

**Issue by vvolas** [lxc/lxd#7782]
LXD rapid restarts 6 times every 0.1 seconds and then gives up.

I doubt this is intentional behavior. 

https://ihatesystemd.com/ugly/#content
https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/

> If you define After=network.target in a Unit file, that means nothing to systemd. And certainly does not mean what you would be likely to expect it to mean:
> 
>     network.target has very little meaning during start-up. It only indicates that the network management stack is up after it has been reached. Whether any network interfaces are already configured when it is reached is undefined. It's [sic] primary purpose is for ordering things properly at shutdown: since the shutdown ordering of units in systemd is the reverse of the startup ordering, any unit that is order After=network.target can be sure that it is stopped before the network is shut down if the system is powered off.

**TLDR I fixed it with adding:
After=network-online.target
Wants=network-online.target**

But it could be fixed adding longer pauses between restarts too, like 10s or even more.

PS. I dont hate systemd, but systemd is confusing, and that site was kinda funny :)

**Comment by stgraber** [comment]
LXD doesn't strictly require network to start up, could you show `journalctl -u snap.lxd.daemon -n 300` to see what caused it to restart?

**Comment by vvolas** [comment]
As I said network is not up. 

systemctl show snap.lxd.daemon.service
RestartUSec=100ms (default)

I think increasing it would be quite harmless and would solve such slow startups.

```
-- Logs begin at Fri 2019-12-20 16:20:35 EET, end at Tue 2020-08-18 06:29:17 EEST. --
Aug 17 14:05:01 universe-linux systemd[1]: snap.lxd.daemon.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 14:05:01 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:01 universe-linux systemd[1]: snap.lxd.daemon.service: Service hold-off time over, scheduling restart.
Aug 17 14:05:01 universe-linux systemd[1]: snap.lxd.daemon.service: Scheduled restart job, restart counter is at 2.
Aug 17 14:05:01 universe-linux systemd[1]: Stopped Service for snap application lxd.daemon.
Aug 17 14:05:01 universe-linux systemd[1]: Started Service for snap application lxd.daemon.
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: => Preparing the system (16740)
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Loading snap configuration
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Setting up mntns symlink (mnt:[4026533303])
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Setting up kmod wrapper
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Preparing /boot
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Preparing a clean copy of /run
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Preparing /run/bin
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Preparing a clean copy of /etc
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Preparing a clean copy of /usr/share/misc
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Setting up ceph configuration
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Setting up LVM configuration
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Rotating logs
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Setting up ZFS (0.8)
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Escaping the systemd cgroups
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ====> Detected cgroup V1
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Escaping the systemd process resource limits
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: ==> Disabling shiftfs on this kernel (auto)
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: => Re-using existing LXCFS
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: => Starting LXD
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: t=2020-08-17T14:05:01+0300 lvl=warn msg=" - Couldn't find the CGroup blkio.weight, I/O weight limits will be ignored"
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: t=2020-08-17T14:05:01+0300 lvl=warn msg=" - Couldn't find the CGroup memory swap accounting, swap limits will be ignored"
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: t=2020-08-17T14:05:01+0300 lvl=eror msg="Failed to start the daemon: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address"
Aug 17 14:05:01 universe-linux lxd.daemon[2072]: Error: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address
Aug 17 14:05:02 universe-linux systemd[1]: snap.lxd.daemon.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 14:05:02 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:03 universe-linux systemd[1]: snap.lxd.daemon.service: Service hold-off time over, scheduling restart.
Aug 17 14:05:03 universe-linux systemd[1]: snap.lxd.daemon.service: Scheduled restart job, restart counter is at 3.
Aug 17 14:05:03 universe-linux systemd[1]: Stopped Service for snap application lxd.daemon.
Aug 17 14:05:03 universe-linux systemd[1]: Started Service for snap application lxd.daemon.
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: => Preparing the system (16740)
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Loading snap configuration
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Setting up mntns symlink (mnt:[4026533303])
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Setting up kmod wrapper
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Preparing /boot
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Preparing a clean copy of /run
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Preparing /run/bin
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Preparing a clean copy of /etc
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Preparing a clean copy of /usr/share/misc
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Setting up ceph configuration
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Setting up LVM configuration
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Rotating logs
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Setting up ZFS (0.8)
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Escaping the systemd cgroups
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ====> Detected cgroup V1
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Escaping the systemd process resource limits
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: ==> Disabling shiftfs on this kernel (auto)
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: => Re-using existing LXCFS
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: => Starting LXD
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: t=2020-08-17T14:05:03+0300 lvl=warn msg=" - Couldn't find the CGroup blkio.weight, I/O weight limits will be ignored"
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: t=2020-08-17T14:05:03+0300 lvl=warn msg=" - Couldn't find the CGroup memory swap accounting, swap limits will be ignored"
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: t=2020-08-17T14:05:03+0300 lvl=eror msg="Failed to start the daemon: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address"
Aug 17 14:05:03 universe-linux lxd.daemon[2243]: Error: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address
Aug 17 14:05:04 universe-linux systemd[1]: snap.lxd.daemon.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 14:05:04 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:04 universe-linux systemd[1]: snap.lxd.daemon.service: Service hold-off time over, scheduling restart.
Aug 17 14:05:04 universe-linux systemd[1]: snap.lxd.daemon.service: Scheduled restart job, restart counter is at 4.
Aug 17 14:05:04 universe-linux systemd[1]: Stopped Service for snap application lxd.daemon.
Aug 17 14:05:04 universe-linux systemd[1]: Started Service for snap application lxd.daemon.
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: => Preparing the system (16740)
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Loading snap configuration
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Setting up mntns symlink (mnt:[4026533303])
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Setting up kmod wrapper
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Preparing /boot
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Preparing a clean copy of /run
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Preparing /run/bin
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Preparing a clean copy of /etc
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Preparing a clean copy of /usr/share/misc
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Setting up ceph configuration
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Setting up LVM configuration
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Rotating logs
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Setting up ZFS (0.8)
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Escaping the systemd cgroups
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ====> Detected cgroup V1
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Escaping the systemd process resource limits
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: ==> Disabling shiftfs on this kernel (auto)
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: => Re-using existing LXCFS
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: => Starting LXD
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: t=2020-08-17T14:05:04+0300 lvl=warn msg=" - Couldn't find the CGroup blkio.weight, I/O weight limits will be ignored"
Aug 17 14:05:04 universe-linux lxd.daemon[2418]: t=2020-08-17T14:05:04+0300 lvl=warn msg=" - Couldn't find the CGroup memory swap accounting, swap limits will be ignored"
Aug 17 14:05:05 universe-linux lxd.daemon[2418]: t=2020-08-17T14:05:05+0300 lvl=eror msg="Failed to start the daemon: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address"
Aug 17 14:05:05 universe-linux lxd.daemon[2418]: Error: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address
Aug 17 14:05:05 universe-linux systemd[1]: snap.lxd.daemon.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 14:05:05 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:06 universe-linux systemd[1]: snap.lxd.daemon.service: Service hold-off time over, scheduling restart.
Aug 17 14:05:06 universe-linux systemd[1]: snap.lxd.daemon.service: Scheduled restart job, restart counter is at 5.
Aug 17 14:05:06 universe-linux systemd[1]: Stopped Service for snap application lxd.daemon.
Aug 17 14:05:06 universe-linux systemd[1]: snap.lxd.daemon.service: Start request repeated too quickly.
Aug 17 14:05:06 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:06 universe-linux systemd[1]: Failed to start Service for snap application lxd.daemon.
Aug 17 14:05:06 universe-linux systemd[1]: Started Service for snap application lxd.daemon.
Aug 17 14:05:06 universe-linux lxd.daemon[2592]: => Preparing the system (16740)
Aug 17 14:05:06 universe-linux lxd.daemon[2592]: ==> Loading snap configuration
Aug 17 14:05:06 universe-linux lxd.daemon[2592]: ==> Setting up mntns symlink (mnt:[4026533303])
Aug 17 14:05:06 universe-linux lxd.daemon[2592]: ==> Setting up kmod wrapper
Aug 17 14:05:06 universe-linux lxd.daemon[2592]: ==> Preparing /boot
Aug 17 14:05:06 universe-linux lxd.daemon[2592]: ==> Preparing a clean copy of /run
Aug 17 14:05:06 universe-linux lxd.daemon[2592]: ==> Preparing /run/bin
Aug 17 14:05:06 universe-linux lxd.daemon[2592]: ==> Preparing a clean copy of /etc
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: ==> Preparing a clean copy of /usr/share/misc
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: ==> Setting up ceph configuration
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: ==> Setting up LVM configuration
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: ==> Rotating logs
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: ==> Setting up ZFS (0.8)
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: ==> Escaping the systemd cgroups
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: ====> Detected cgroup V1
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: ==> Escaping the systemd process resource limits
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: ==> Disabling shiftfs on this kernel (auto)
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: => Re-using existing LXCFS
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: => Starting LXD
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: t=2020-08-17T14:05:07+0300 lvl=warn msg=" - Couldn't find the CGroup blkio.weight, I/O weight limits will be ignored"
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: t=2020-08-17T14:05:07+0300 lvl=warn msg=" - Couldn't find the CGroup memory swap accounting, swap limits will be ignored"
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: t=2020-08-17T14:05:07+0300 lvl=eror msg="Failed to start the daemon: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address"
Aug 17 14:05:07 universe-linux lxd.daemon[2592]: Error: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address
Aug 17 14:05:08 universe-linux systemd[1]: snap.lxd.daemon.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 14:05:08 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:08 universe-linux systemd[1]: snap.lxd.daemon.service: Service hold-off time over, scheduling restart.
Aug 17 14:05:08 universe-linux systemd[1]: snap.lxd.daemon.service: Scheduled restart job, restart counter is at 1.
Aug 17 14:05:08 universe-linux systemd[1]: Stopped Service for snap application lxd.daemon.
Aug 17 14:05:08 universe-linux systemd[1]: Started Service for snap application lxd.daemon.
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: => Preparing the system (16740)
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Loading snap configuration
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Setting up mntns symlink (mnt:[4026533303])
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Setting up kmod wrapper
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Preparing /boot
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Preparing a clean copy of /run
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Preparing /run/bin
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Preparing a clean copy of /etc
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Preparing a clean copy of /usr/share/misc
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Setting up ceph configuration
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Setting up LVM configuration
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Rotating logs
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Setting up ZFS (0.8)
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Escaping the systemd cgroups
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ====> Detected cgroup V1
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Escaping the systemd process resource limits
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: ==> Disabling shiftfs on this kernel (auto)
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: => Re-using existing LXCFS
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: => Starting LXD
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: t=2020-08-17T14:05:08+0300 lvl=warn msg=" - Couldn't find the CGroup blkio.weight, I/O weight limits will be ignored"
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: t=2020-08-17T14:05:08+0300 lvl=warn msg=" - Couldn't find the CGroup memory swap accounting, swap limits will be ignored"
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: t=2020-08-17T14:05:08+0300 lvl=eror msg="Failed to start the daemon: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address"
Aug 17 14:05:08 universe-linux lxd.daemon[2759]: Error: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address
Aug 17 14:05:09 universe-linux systemd[1]: snap.lxd.daemon.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 14:05:09 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:09 universe-linux systemd[1]: snap.lxd.daemon.service: Service hold-off time over, scheduling restart.
Aug 17 14:05:09 universe-linux systemd[1]: snap.lxd.daemon.service: Scheduled restart job, restart counter is at 2.
Aug 17 14:05:09 universe-linux systemd[1]: Stopped Service for snap application lxd.daemon.
Aug 17 14:05:09 universe-linux systemd[1]: Started Service for snap application lxd.daemon.
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: => Preparing the system (16740)
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Loading snap configuration
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Setting up mntns symlink (mnt:[4026533303])
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Setting up kmod wrapper
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Preparing /boot
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Preparing a clean copy of /run
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Preparing /run/bin
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Preparing a clean copy of /etc
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Preparing a clean copy of /usr/share/misc
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Setting up ceph configuration
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Setting up LVM configuration
Aug 17 14:05:09 universe-linux lxd.daemon[2925]: ==> Rotating logs
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: ==> Setting up ZFS (0.8)
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: ==> Escaping the systemd cgroups
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: ====> Detected cgroup V1
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: ==> Escaping the systemd process resource limits
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: ==> Disabling shiftfs on this kernel (auto)
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: => Re-using existing LXCFS
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: => Starting LXD
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: t=2020-08-17T14:05:10+0300 lvl=warn msg=" - Couldn't find the CGroup blkio.weight, I/O weight limits will be ignored"
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: t=2020-08-17T14:05:10+0300 lvl=warn msg=" - Couldn't find the CGroup memory swap accounting, swap limits will be ignored"
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: t=2020-08-17T14:05:10+0300 lvl=eror msg="Failed to start the daemon: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address"
Aug 17 14:05:10 universe-linux lxd.daemon[2925]: Error: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address
Aug 17 14:05:11 universe-linux systemd[1]: snap.lxd.daemon.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 14:05:11 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:11 universe-linux systemd[1]: snap.lxd.daemon.service: Service hold-off time over, scheduling restart.
Aug 17 14:05:11 universe-linux systemd[1]: snap.lxd.daemon.service: Scheduled restart job, restart counter is at 3.
Aug 17 14:05:11 universe-linux systemd[1]: Stopped Service for snap application lxd.daemon.
Aug 17 14:05:11 universe-linux systemd[1]: Started Service for snap application lxd.daemon.
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: => Preparing the system (16740)
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Loading snap configuration
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Setting up mntns symlink (mnt:[4026533303])
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Setting up kmod wrapper
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Preparing /boot
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Preparing a clean copy of /run
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Preparing /run/bin
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Preparing a clean copy of /etc
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Preparing a clean copy of /usr/share/misc
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Setting up ceph configuration
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Setting up LVM configuration
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Rotating logs
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Setting up ZFS (0.8)
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Escaping the systemd cgroups
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ====> Detected cgroup V1
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Escaping the systemd process resource limits
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: ==> Disabling shiftfs on this kernel (auto)
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: => Re-using existing LXCFS
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: => Starting LXD
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: t=2020-08-17T14:05:11+0300 lvl=warn msg=" - Couldn't find the CGroup blkio.weight, I/O weight limits will be ignored"
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: t=2020-08-17T14:05:11+0300 lvl=warn msg=" - Couldn't find the CGroup memory swap accounting, swap limits will be ignored"
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: t=2020-08-17T14:05:11+0300 lvl=eror msg="Failed to start the daemon: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address"
Aug 17 14:05:11 universe-linux lxd.daemon[3089]: Error: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address
Aug 17 14:05:12 universe-linux systemd[1]: snap.lxd.daemon.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 14:05:12 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:12 universe-linux systemd[1]: snap.lxd.daemon.service: Service hold-off time over, scheduling restart.
Aug 17 14:05:12 universe-linux systemd[1]: snap.lxd.daemon.service: Scheduled restart job, restart counter is at 4.
Aug 17 14:05:12 universe-linux systemd[1]: Stopped Service for snap application lxd.daemon.
Aug 17 14:05:12 universe-linux systemd[1]: Started Service for snap application lxd.daemon.
Aug 17 14:05:12 universe-linux lxd.daemon[3259]: => Preparing the system (16740)
Aug 17 14:05:12 universe-linux lxd.daemon[3259]: ==> Loading snap configuration
Aug 17 14:05:12 universe-linux lxd.daemon[3259]: ==> Setting up mntns symlink (mnt:[4026533303])
Aug 17 14:05:12 universe-linux lxd.daemon[3259]: ==> Setting up kmod wrapper
Aug 17 14:05:12 universe-linux lxd.daemon[3259]: ==> Preparing /boot
Aug 17 14:05:12 universe-linux lxd.daemon[3259]: ==> Preparing a clean copy of /run
Aug 17 14:05:12 universe-linux lxd.daemon[3259]: ==> Preparing /run/bin
Aug 17 14:05:12 universe-linux lxd.daemon[3259]: ==> Preparing a clean copy of /etc
Aug 17 14:05:12 universe-linux lxd.daemon[3259]: ==> Preparing a clean copy of /usr/share/misc
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: ==> Setting up ceph configuration
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: ==> Setting up LVM configuration
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: ==> Rotating logs
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: ==> Setting up ZFS (0.8)
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: ==> Escaping the systemd cgroups
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: ====> Detected cgroup V1
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: ==> Escaping the systemd process resource limits
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: ==> Disabling shiftfs on this kernel (auto)
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: => Re-using existing LXCFS
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: => Starting LXD
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: t=2020-08-17T14:05:13+0300 lvl=warn msg=" - Couldn't find the CGroup blkio.weight, I/O weight limits will be ignored"
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: t=2020-08-17T14:05:13+0300 lvl=warn msg=" - Couldn't find the CGroup memory swap accounting, swap limits will be ignored"
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: t=2020-08-17T14:05:13+0300 lvl=eror msg="Failed to start the daemon: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address"
Aug 17 14:05:13 universe-linux lxd.daemon[3259]: Error: Bind network address: listen tcp 10.0.199.210:8443: bind: cannot assign requested address
Aug 17 14:05:14 universe-linux systemd[1]: snap.lxd.daemon.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 14:05:14 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:14 universe-linux systemd[1]: snap.lxd.daemon.service: Service hold-off time over, scheduling restart.
Aug 17 14:05:14 universe-linux systemd[1]: snap.lxd.daemon.service: Scheduled restart job, restart counter is at 5.
Aug 17 14:05:14 universe-linux systemd[1]: Stopped Service for snap application lxd.daemon.
Aug 17 14:05:14 universe-linux systemd[1]: snap.lxd.daemon.service: Start request repeated too quickly.
Aug 17 14:05:14 universe-linux systemd[1]: snap.lxd.daemon.service: Failed with result 'exit-code'.
Aug 17 14:05:14 universe-linux systemd[1]: Failed to start Service for snap application lxd.daemon.
-- Reboot --

```

**Comment by tomponline** [comment]
So the issue looks like you've set a LXD bind address on a specific IP address, which explains why it can't listen on that address until the network is online.

You may also be able to workaround this issue by allowing non-local binds https://www.cyberciti.biz/faq/linux-bind-ip-that-doesnt-exist-with-net-ipv4-ip_nonlocal_bind/

**Comment by stgraber** [comment]
That would cause other issues though. I think we should go for a 1min retry loop when binding addresses, then if it's core.https_address, we move on, if it's cluster.https_address, we fail.
